src/polly/src/event_manager.rs

```diff
@@ -178,14 +178,15 @@ impl EventManager {
     /// Wait for events for a maximum timeout of `miliseconds`. Dispatch the events to the
     /// registered signal handlers.
     pub fn run_with_timeout(&mut self, milliseconds: i32) -> Result<usize> {
-        let event_count = self
-            .epoll
-            .wait(
-                EventManager::EVENT_BUFFER_SIZE,
-                milliseconds,
-                &mut self.ready_events[..],
-            )
-            .map_err(Error::Poll)?;
+        let event_count = match self.epoll.wait(
+            EventManager::EVENT_BUFFER_SIZE,
+            milliseconds,
+            &mut self.ready_events[..],
+        ) {
+            Ok(event_count) => event_count,
+            Err(e) if e.raw_os_error() == Some(libc::EINTR) => 0,
```

Comment:
We should just return here. There is no need to call `dispatch_events`.

Reply:
The same _separation of concerns paradigm_ vs _optimizing an unlikely case_ could be applied here as well, but since this is a very simple and clear function I think it's perfectly fine to short-circuit it.

Reply:
I wanted to suggest avoiding the dispatch_events extra call for the unlikely usecase, but I think @acatangiu is in point here.

Reply:
For this code path I guess is fine to not return 0, but in the `micro_http` case, you have to read quite a lot of code to see that if you receive an interrupt an empty vector is returned.

From my point of view it actually makes the code harder to read, not easier. The separation of concerns main advantage is code readability and maintainability. It typically applies to components, not to how code is structured in a single function. Both of my comments were nits, so I think we can merge the code how it is now.